lib/SMTLIB2/Solver.cpp

```diff
@@ -89,16 +89,35 @@ struct SMTLIBParser {
     return false;
   }
 
-  APInt parseModel(std::string &ErrStr) {
-    if (!consumeExpected('(', ErrStr))
-      return APInt();
-    if (!consumeExpected('(', ErrStr))
+  bool consumeBvName(std::string &ErrStr) {
+    while (Begin != End && (*Begin != ' ' && *Begin != '\n' && *Begin != '\r' &&
+                            *Begin != '\t'))
+      ++Begin;
+    if (Begin == End) {
+      ErrStr = "unexpected EOF";
+      return false;
+    }
+    return true;
+  }
+
+  APInt parseBinaryBitVector(std::string &ErrStr) {
+    ErrStr.clear();
```

Comment:
``b`` is being consumed while checking for ``x`` before moving to ``parseBinaryVector``. I agree that we should explicitly check if the consumed token was ``b`` and not something else.